arch/risc-v/src/common/riscv_macros.S

```diff
@@ -106,11 +106,6 @@
   and        t2, t0, t1
   li         t1, MSTATUS_FS_DIRTY
   bne        t2, t1, 1f
```

Comment:
blt, you need to save when status is in {DIRTY, CLEAN}

Reply:
In fact, the only line of code that needs to be changed for this patch to work correctly, is this bne -> blt 

That achieves what we need for now:
save FPU always when status is not in {OFF, INIT}
restore FPU always when status is in {DIRTY, CLEAN}

Reply:
should we change to:
```
  li         t1, MSTATUS_FS_INIT
  ble        t2, t1, 1f
```

Reply:
Here, the functions of bne and blt should be the same, right?

Reply:
no, MSTATUS_FS_CLEAN need save too